clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -4895,7 +4895,7 @@ void Sema::InstantiateFunctionDefinition(SourceLocation PointOfInstantiation,
   // Never implicitly instantiate a builtin; we don't actually need a function
   // body.
   if (Function->getBuiltinID() && TSK == TSK_ImplicitInstantiation &&
-      !DefinitionRequired)
+      !DefinitionRequired && !Function->getReturnType()->isUndeducedType())
```

Comment:
I think this ends up affecting more builtins than just the ones we care about here. I’m not sure that’s a good idea, but I’m also not too familiar w/ builtins.